packages/amplify-cli/src/cli.js

```diff
@@ -16,10 +17,14 @@ async function run(argv) {
     .version() // provides default for version, v, --version, -v
     .create();
 
+  if (argv[2] !== MIGRATE) {
+    const yesFlag = argv.includes('-y') || argv.includes('--yes');
```

Comment:
I think providing a --yes flag support for this could be dangerous since this would bypass the critical questions we ask the users about the migration (since this involves a risk of them losing their current changes to files). For eg: If someone runs `amplify push --yes`, the migration shouldn't trigger automatically and the users need to answer the migration questions. We should be careful about this since its a special case.

Reply:
I see your point. 
The other side of this is that, without the yes flag, migration would normally need to be run with human interaction, and it becomes inconvenient to run it within a CICD script.